### PR TITLE
refac: Publish events from results in delta table

### PR DESCRIPTION
**Pull request by BjarkeMeier**
## References

* #447

### CODE REVIEW

File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/CalculationResults/CalculationResultQueries.cs [hunk 1]
```diff
@@ -97,7 +102,7 @@ private static TimeSeriesPoint CreateTimeSeriesPoint(SqlResultRow row)
     }
 
     private static CalculationResult CreateCalculationResult(
```
Comment (JesperGomes) [inline source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/CalculationResults/CalculationResultQueries.cs:104]: Candidate for factory method?

Comment (BjarkeMeier) [reply source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/CalculationResults/CalculationResultQueries.cs:104]: I'm not sure. It's using the one simple ctor of calculation result. Would you like to add a dependency to SqlResultRow in CalculationResult? I would prefer not.

File: source/dotnet/wholesale-api/Events/Events.Application/CalculationResultPublishing/CalculationResultPublisher.cs [hunk 2]
```diff
@@ -12,191 +12,75 @@
...
-                        batchGridAreaInfo.BatchId,
-                        batchGridAreaInfo.GridAreaCode,
-                        timeSeriesType,
-                        energySupplier.Gln,
-                        brp.Gln)
-                    .ConfigureAwait(false);
-
-                var integrationEvent = _calculationResultCompletedFactory.CreateForEnergySupplierByBalanceResponsibleParty(result, batchGridAreaInfo, energySupplier.Gln, brp.Gln);
-                await _integrationEventPublisher.PublishAsync(integrationEvent).ConfigureAwait(false);
+                await PublishForGridAccessProviderAsync(calculationResult).ConfigureAwait(false);
```
Comment (JesperGomes) [inline source/dotnet/wholesale-api/Events/Events.Application/CalculationResultPublishing/CalculationResultPublisher.cs:44]: I assume we still publish only certain time series types - I'm just not sure where this logic has moved?

File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/CalculationResults/CalculationResultQueries.cs (hunk 1 above)
Comment (JesperGomes) [reply source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/CalculationResults/CalculationResultQueries.cs:104]: Maybe not. I'm fine with this 👍

File: source/dotnet/wholesale-api/Events/Events.Application/CalculationResultPublishing/CalculationResultPublisher.cs (hunk 2 above)
Comment (BjarkeMeier) [reply source/dotnet/wholesale-api/Events/Events.Application/CalculationResultPublishing/CalculationResultPublisher.cs:44]: Why should we only publish certain time series types? That would sound like a very EDI specific responsibility/knowledge in the wholesale domain.

Comment (JesperGomes) [reply source/dotnet/wholesale-api/Events/Events.Application/CalculationResultPublishing/CalculationResultPublisher.cs:44]: It's not a problem as such. It's just a change in behavior. But EDI should be agnostic to that.

Comment (BjarkeMeier) [reply source/dotnet/wholesale-api/Events/Events.Application/CalculationResultPublishing/CalculationResultPublisher.cs:44]: I think this (new) approach is more correct. Do you have a different perspective?

Comment (JesperGomes) [reply source/dotnet/wholesale-api/Events/Events.Application/CalculationResultPublishing/CalculationResultPublisher.cs:44]: Not really. The only issue I can think of is performance, but I guess it is okay that we send them although many of the time series types are not consumed by anyone.

Comment (BjarkeMeier) [reply source/dotnet/wholesale-api/Events/Events.Application/CalculationResultPublishing/CalculationResultPublisher.cs:44]: Sounds like I need an update. What (many) time series types are we talking about?
Are we calculating a lot of results that should never be sent? (Which?)

**Review by johevemi — APPROVED** [review]

File: source/dotnet/wholesale-api/Events/Events.Application/CalculationResultPublishing/CalculationResultPublisher.cs (hunk 2 above)
Comment (johevemi) [reply source/dotnet/wholesale-api/Events/Events.Application/CalculationResultPublishing/CalculationResultPublisher.cs:44]: I think we should just send all time series type for now

**Pull request merged**